lib/Myriad/Bootstrap.pm

```diff
@@ -128,35 +175,49 @@ sub boot {
         }
     }
 
-    # Establish comms channel for child process
-    socketpair my $child_pipe, my $parent_pipe, $constant{AF_UNIX}, $constant{SOCK_STREAM}, $constant{PF_UNSPEC}
-        or die $!;
+    my ($inotify_parent_pipe, $inotify_child_pipe) = open_pipe();
 
-    { # Unbuffered writes
-        my $old = select($child_pipe);
-        $| = 1; select($parent_pipe);
-        $| = 1; select($old);
+    {
+        if (my $pid = fork // die "fork! $!") {
```

Comment:
Do you need `$inotify_parent_pipe` in the parent? If not, closing is a good idea.